pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -578,3 +582,292 @@ func TestFlowAggregator_closeUpdateChBeforeFlowExportLoopReturns(t *testing.T) {
+	require.NoError(t, err)
+
+	err = fa.InitAggregationProcess()
+	require.NoError(t, err)
+}
+
+func TestFlowAggregator_fillK8sMetadata(t *testing.T) {
+	dummySrcPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "dummySourcePod",
+		},
+		Spec: v1.PodSpec{
+			NodeName: "dummySourceNode",
+		},
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "192.168.1.2",
+				},
+			},
+		},
+	}
+	dummyDstPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "dummyDestinationPod",
+		},
+		Spec: v1.PodSpec{
+			NodeName: "dummyDestinationNode",
+		},
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "192.168.1.3",
+				},
+			},
+		},
+	}
+	emptyStr := make([]byte, 0)
+	sourcePodNameElem, err := ipfixentities.DecodeAndCreateInfoElementWithValue(ipfixentities.NewInfoElement("sourcePodName", uint16(0), ipfixentities.String, ipfixregistry.AntreaEnterpriseID, uint16(0)), emptyStr)
+	require.NoError(t, err)
+	sourcePodNamespaceElem, err := ipfixentities.DecodeAndCreateInfoElementWithValue(ipfixentities.NewInfoElement("sourcePodNamespace", uint16(0), ipfixentities.String, ipfixregistry.AntreaEnterpriseID, uint16(0)), emptyStr)
+	require.NoError(t, err)
+	sourceNodeNameElem, err := ipfixentities.DecodeAndCreateInfoElementWithValue(ipfixentities.NewInfoElement("sourceNodeName", uint16(0), ipfixentities.String, ipfixregistry.AntreaEnterpriseID, uint16(0)), emptyStr)
+	require.NoError(t, err)
+	destinationPodNameElem, err := ipfixentities.DecodeAndCreateInfoElementWithValue(ipfixentities.NewInfoElement("destinationPodName", uint16(0), ipfixentities.String, ipfixregistry.AntreaEnterpriseID, uint16(0)), emptyStr)
+	require.NoError(t, err)
+	destinationPodNamespaceElem, err := ipfixentities.DecodeAndCreateInfoElementWithValue(ipfixentities.NewInfoElement("destinationPodNamespace", uint16(0), ipfixentities.String, ipfixregistry.AntreaEnterpriseID, uint16(0)), emptyStr)
+	require.NoError(t, err)
+	destinationNodeNameElem, err := ipfixentities.DecodeAndCreateInfoElementWithValue(ipfixentities.NewInfoElement("destinationNodeName", uint16(0), ipfixentities.String, ipfixregistry.AntreaEnterpriseID, uint16(0)), emptyStr)
+	require.NoError(t, err)
+
+	ctrl := gomock.NewController(t)
+	mockRecord := ipfixentitiestesting.NewMockRecord(ctrl)
+	client := fake.NewSimpleClientset()
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	informerFactory.Core().V1().Pods().Informer().AddIndexers(cache.Indexers{podInfoIndex: podInfoIndexFunc})
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	informerFactory.Start(stopCh)
+	informerFactory.Core().V1().Pods().Informer().GetIndexer().Add(dummySrcPod)
+	informerFactory.Core().V1().Pods().Informer().GetIndexer().Add(dummyDstPod)
+
+	ipv4Key := ipfixintermediate.FlowKey{
+		SourceAddress:      "192.168.1.2",
+		DestinationAddress: "192.168.1.3",
+	}
+
+	t.Run("pod metadata and node name", func(t *testing.T) {
```

Comment:
no need for a subtest if you have a single test

Reply:
removed.